### DISCUSSION TITLE
HAproxy config recreated automatically

### DISCUSSION THREAD

**Discussion by mishanchus** [autobase-tech/autobase#595] [Q&A] [answered]
Hello,
I'm trying to change some parameters in /etc/haproxy/haproxy.cfg, but after a while I discover that the file is being recreated with the previous settings. Why is this happening?

**Comment by vitabaks** [comment, marked as answer]
Hello @mishanchus 

because the configuration is managed by the confd service.

> **Components of load balancing:**
> 
> [HAProxy](http://www.haproxy.org/) is a free, very fast and reliable solution offering high availability, load balancing, and proxying for TCP and HTTP-based applications.
> 
> [confd](https://github.com/kelseyhightower/confd) manage local application configuration files using templates and data from etcd or consul. **Used to automate HAProxy configuration file management.**
> 
> [Keepalived](https://github.com/acassen/keepalived) (optional, if the cluster_vip variable is specified) provides a virtual high-available IP address (VIP) and single entry point for databases access. Implementing VRRP (Virtual Router Redundancy Protocol) for Linux. In our configuration keepalived checks the status of the HAProxy service and in case of a failure delegates the VIP to another server in the cluster.

In theory, you can change the parameter with your hands, but you need to do it in the template along the path `/etc/confd/templates/haproxy.tmpl`

**Reply by vitabaks** [reply]
I could add an option in which it would be possible to exclude `confd` from the installation if, for example, you want to manage the HAProxy configuration yourself (for example, when scaling a cluster or disabling one of the servers). 

Let me know if you are interested in this option.

**Reply by mishanchus** [reply]
Thank you, Vitaly! I don’t know about the option, for now it was important for me to understand the reasons and change the config.
